### PR TITLE
Selectively enable batch RPC providers

### PR DESCRIPTION
**Pull request by hyphenized**
Switches to JsonRPCBatchProvider for both ankr and 1rpc. This is superb because:

- Quotas and limits are set per request as opposed to per call.
- Fixes to an extent the thundering herd issue related to enrichment which was causing 429s

Although ethers provides support for batch rpc providers, it does not handle well responses where the results are not ordered. As this is in compliance with the spec, some manual patching is necessary to process these responses correctly.

## To Test
Checking that rpc network requests are being batched and, there are no errors or inconsistent behaviour within the wallet should be enough. E2E tests passing should also give us more confidence that this is working correctly.

- [x] Open the network inspector.
- [x] Onboard an account.
- [x] Verify that outgoing rpc network requests are sometimes bundled together.

Latest build: [extension-builds-3580](https://github.com/tahowallet/extension/suites/14720000395/artifacts/836157163) (as of Tue, 01 Aug 2023 08:12:18 GMT).

### CODE REVIEW

**Review by kkosiorowska — COMMENTED** [review]
I have tested it and seems that work good. Just minor question in comment.

File: background/services/chain/serial-fallback-provider.ts [hunk 1]
```diff
@@ -1005,6 +1006,10 @@ function getProviderCreator(
     return new WebSocketProvider(rpcUrl)
   }
 
+  if (/rpc\.ankr\.com|1rpc\.io/.test(url.hostname)) {
```
[the remark this answers is not in the document]
Comment (hyphenized) [reply background/services/chain/serial-fallback-provider.ts]: Yup, thanks for sharing!

**Review by kkosiorowska — APPROVED** [review]
✅

**Comment by jagodarybacka** [comment]
@hyphenized seems like using `JsonRpcBatchProvider` has accidentally fixed issue on Velodrome we had for a looong time https://github.com/tahowallet/extension/issues/2216 - do you think it would make sense to use batch provider by default?

**Comment by hyphenized** [comment]
I need to dig a bit into this but that seems like a different type of issue 👀

**Pull request merged**